src/test/java/spoon/test/generics/GenericsTest.java

```diff
@@ -1036,4 +1036,38 @@ public void testDiamondComplexGenericsRxJava() {
 
 		assertTrue(invocationDetected);
 	}
+
+	@Test
+	public void testGetDeclarationOfTypeParameterReference() {
+		Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/java/spoon/test/generics/testclasses/ExtendedPaella.java");
+		launcher.addInputResource("./src/test/java/spoon/test/generics/testclasses/Paella.java");
+		launcher.buildModel();
+
+		Factory factory = launcher.getFactory();
+
+		CtClass extendedPaella = factory.getModel().getElements(new NameFilter<CtClass>("ExtendedPaella")).get(0);
+		List<CtTypeParameter> typeParameterList = extendedPaella.getFormalCtTypeParameters();
+
+		assertEquals(1, typeParameterList.size());
+
+		CtMethod totoMethod = factory.getModel().getElements(new NameFilter<CtMethod>("toto")).get(0);
+		CtTypeReference returnTypeToto = totoMethod.getType();
+
+		CtType declaration = returnTypeToto.getDeclaration();
+
+		assertTrue(declaration.equals(typeParameterList.get(0)));
```

Comment:
I preffer asserSame instead assertEquals in this case. May be there might exist a equal type parameter, whis is not correct one.